### PR TITLE
CloudMonitoring: Prevent resource type variable function from crashing

### LINKED ISSUE #30491: Cloud monitoring: Unable to create new resource type template variable
**What happened**:
I get the following UI errors:
`Cannot read property 'map' of undefined`
`Error updating options: Cannot read property 'map' of undefined`
In the console I get [this](https://gist.github.com/papagian/2f96e3845c0fef4494c4f3a6b5ebcb3e) error (grafana version 7.3.0).

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
- Create a new template query variable
- Select a cloud monitoring datasource
- Select `Resource type` as query type

**Anything else we need to know?**:

**Environment**:
- Grafana version: master, 7.3.6 and 7.3.0
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by sunker**
Fixes #30491

### CODE REVIEW

**Review by papagian — APPROVED** [review]
I think it works fine: if I select `Resource Type` as a query type no more crashes (but I don't get any result). If I use another cloud monitoring datasource, it still does not crashes but I get a permission denied error (I guess because I lack permission to perform such query).

File: public/app/plugins/datasource/cloud-monitoring/CloudMonitoringMetricFindQuery.ts [hunk 1]
```diff
@@ -112,7 +112,7 @@ export default class CloudMonitoringMetricFindQuery {
     }
     const refId = 'handleResourceTypeQueryQueryType';
     const labels = await this.datasource.getLabels(selectedMetricType, refId, projectName);
-    return labels['resource.type'].map(this.toFindQueryResult);
+    return labels.hasOwnProperty('resource.type') ? labels['resource.type'].map(this.toFindQueryResult) : [];
```
Comment (joshhunt) [inline public/app/plugins/datasource/cloud-monitoring/CloudMonitoringMetricFindQuery.ts:115]: you could probably write this as `labels['resource.type']?.map() ?? []` if you so cared

**Review by joshhunt — APPROVED** [review]
looks good!

File: public/app/plugins/datasource/cloud-monitoring/CloudMonitoringMetricFindQuery.ts (hunk 1 above)
Comment (joshhunt) [reply public/app/plugins/datasource/cloud-monitoring/CloudMonitoringMetricFindQuery.ts:115]: oh I didn't mean to actually submit this comment 😅

Comment (sunker) [reply public/app/plugins/datasource/cloud-monitoring/CloudMonitoringMetricFindQuery.ts]: yes yes much better, thanks!

**Pull request merged**